WordPress/gutenberg
# Block Support: Separate opt in for font style and weight options

## Description
Updates the recently added font appearance block support to separate the style and weight aspects in terms of opting in via block.json. 

This includes:
* Updating the detection of support opt-in within typography tools
* Updating the UI control to handle the different opt-in situations
* Adding translation for the combined UI option labels allowing order to be switched if appropriate for a locale
* Extracting concerns of how the selections are stored within style attributes from the UI control

## How has this been tested?
Manually.

#### Testing Instructions
1. Check out this PR
2. Edit a post adding a navigation block
3. Select the navigation block and ensure the ability to select a font style and weight combination still works
4. Update `packages/block-library/src/navigation/block.json` toggling `__experimentalFontStyle` off in the supports array
5. Reload the editor( you may need to restart dev env to pickup the changed block.json )
6. Confirm the navigation block displays a "Font weight" option instead of the previous "Appearance" option and it works
7. Update the `block.json` again toggling on `__experimentalFontStyle` and toggling off `__experimentalFontWeight`
8. Confirm the navigation block displays a "Font style" option and it works
9. Finally, test removing or toggling off both options and confirm there are no errors and no font style/weight option is displayed.

## Screenshots 
| Style & Weight | Style Only | Weight Only |
|------|-----------|-------------|
| ![FontStyleWeight1](https://user-images.githubusercontent.com/60436221/98666738-2b65ea80-2399-11eb-9c8b-fd18c3dee34e.gif) | ![FontStyleWeight2](https://user-images.githubusercontent.com/60436221/98666752-33258f00-2399-11eb-904b-fd7492207c6f.gif) | ![FontStyleWeight3](https://user-images.githubusercontent.com/60436221/98666797-40427e00-2399-11eb-99a6-dea01d800b68.gif) |

## Types of changes
Breaking change. 

This changes the block support flags for the font style/weight feature that was recently merged but not yet part of a release.

## Checklist:
